webui/react/src/pages/TrialDetails.tsx

```diff
@@ -15,23 +15,68 @@ import usePolling from 'hooks/usePolling';
 import useRestApi from 'hooks/useRestApi';
 import TrialActions, { Action as TrialAction } from 'pages/TrialDetails/TrialActions';
 import TrialInfoBox from 'pages/TrialDetails/TrialInfoBox';
+import { routeAll } from 'routes';
+import { forkExperiment } from 'services/api';
 import { getExperimentDetails, getTrialDetails, isNotFound } from 'services/api';
 import { TrialDetailsParams } from 'services/types';
-import { ExperimentDetails, TrialDetails } from 'types';
+import { ExperimentDetails, RawJson, TrialDetails } from 'types';
 import { clone } from 'utils/data';
 import { trialHParamsToExperimentHParams } from 'utils/types';
 
 interface Params {
   trialId: string;
 }
 
+const getTrialLength = (experimentConfig: RawJson): [string, number] => {
+  const maxLength = experimentConfig.searcher.max_length;
+  if (!maxLength) {
+    // provide backward compat for max_steps
+    let maxSteps = experimentConfig.searcher.max_steps;
+    maxSteps = parseInt(maxSteps) || 100;
+    const batchesPerStep = experimentConfig.batches_per_step || 100;
+    return [ 'batches', maxSteps * batchesPerStep ];
+  }
+  return Object.entries(maxLength)[0] as [string, number];
+};
+
+const setTrialLength = (experimentConfig: RawJson, length: number): void => {
+  const lengthType = getTrialLength(experimentConfig)[0];
+  experimentConfig.searcher.max_length = { [lengthType]: length } ;
+  // in case we are dealing with an old config
+  delete experimentConfig.searcher.max_steps;
+};
+
+const trialContinueConfig =
+(experimentConfig: RawJson, trialHparams: Record<string, string>, trialId: number): RawJson => {
+  const newSearcher = {
+    max_length: experimentConfig.searcher.max_length,
+    metric: experimentConfig.searcher.metric,
+    name: experimentConfig.searcher.name,
+    smaller_is_better: experimentConfig.searcher.smaller_is_better,
+    source_trial_id: trialId,
+  };
+  const newHyperparameters = trialHParamsToExperimentHParams(trialHparams);
+
+  return {
+    ...experimentConfig,
+    description: experimentConfig.description,
+    hyperparameters: newHyperparameters,
+    searcher: newSearcher,
+  };
+};
+
 const TrialDetailsComp: React.FC = () => {
   const { trialId: trialIdParam } = useParams<Params>();
   const trialId = parseInt(trialIdParam);
   const [ trial, triggerTrialRequest ] =
     useRestApi<TrialDetailsParams, TrialDetails>(getTrialDetails, { id: trialId });
   const [ contModalVisible, setContModalVisible ] = useState(false);
   const [ contModalConfig, setContModalConfig ] = useState('Loading');
+  const [ contFormVisible, setContFormVisible ] = useState<boolean>(false);
+  const [ contMaxLength, setContMaxLength ] = useState<number>(100);
```

Comment:
**question:** I asked about whether there is a good default value to set to for max length and the answer was "not really". Can we leave it blank so it forces the user to have to make that decision. @vishnu2kmohan @stoksc what are your thoughts on the default value of 100 for max length / batches?

Reply:
Yeah, I don't think there's a great default for how long to continue the trial. 100 batches is probably just fine.

Reply:
What if the user specified records and epochs instead of batches in the base trial?
The user would expect that the unit from the base trial be carried over?

Reply:
If we want that, then maybe if records, `100 * global_batch_size`, if batches , `100` and if epochs `math.ceil((100 * global_batch_size) / records_per_epoch)`. But in the event `global_batch_size`, we pick the `max`?

Reply:
But I think just not having a default is better, at that point.

Reply:
> Can we leave it blank

yes, and it'd fall back to what the experiment had set before which is the same behavior as the existing trial page. so I thnk this should be fine. does this sound right?